### PR TITLE
Add the ability to reverse iterate over the game list

### PR DESCRIPTION
**Pull request by JamyDev**
Useful for the newest filter so we can search back to front.

### CODE REVIEW

File: game.go [hunk 1]
```diff
@@ -37,11 +38,12 @@ func (c *Client) GameList(queries ...Query) (*GameListResult, error) {
 
 // Returns a channel of games (and errors) which can be iterated
 // over transparently, without manually dealing with pagination.
-func (c *Client) GameListIterate(queries ...Query) *GameListIterator {
+func (c *Client) GameListIterate(reverse bool, queries ...Query) *GameListIterator {
```
Comment (ttaylorr) [inline game.go]: How about a new `type` called direction?

``` go
type Direction int

const (
        Forward Direction = iota
        Backward
)

func (c *Client) GameListIterate(d Direction, queries ...Query) *GameListIterator { ... }
```

Comment (JamyDev) [inline game.go]: If that makes more sense in the Go spectrum, sure :)

Comment (ttaylorr) [inline game.go]: Direction could also have a method `Next` which handles all of the logic on L71-L87.  I'm thinking something like:

``` go
type Direction int

func (d Direction) Next(pager Pager) int {
        switch d {
        case Forward:
                if pager.Limit > pager.From {
                        return 0
                }

                return pager.From - pager.Limit
        case Backward:
                // ...
        }

        return -1
}
```

Comment (connor4312) [inline game.go]: A type for direction would definitely be more idiomatic :wink:

Comment (connor4312) [inline game.go]: Method also looks like it could be nice.

Comment (JamyDev) [inline game.go]: Yup, I'll push these changes within the next hour

**Comment by JamyDev** [comment]
Changes pushed, tests pass. However travis is broken due to the switch from MCPH to WatchBeam. If someone could set up coveralls for this repo again? :)

File: readme.md [hunk 2]
```diff
@@ -23,7 +23,9 @@ func main() {
 
     // Even iterate over results. This will continue requesting
     // additional pages transparently.
-    it := player.New().GameListIterate()
+    // Optionally specify the direction to go (Forward or Backward)
+    // starting from your _from value
+    it := player.New().GameListIterate(Forward)
```
Comment (connor4312) [inline readme.md]: player.Forward

File: game.go [hunk 3]
```diff
@@ -25,8 +25,42 @@ type GameListResult struct {
...
+    Forward Direction = iota
+    Backward
+)
+
+func (d Direction) Next(pager Pager) int {
+		var next int
+        switch d {
+        case Forward:
+                next = pager.Limit + pager.From
+				if next >= pager.Total {
```
Comment (ttaylorr) [inline game.go]: This indentation is really weird. Did you run this through `go fmt`?

File: game.go [hunk 4]
```diff
@@ -25,8 +25,42 @@ type GameListResult struct {
...
 
+type Direction int
+
+const (
+    Forward Direction = iota
+    Backward
+)
+
+func (d Direction) Next(pager Pager) int {
+		var next int
```
Comment (ttaylorr) [inline game.go]: You can kill this function var and define them inside of each case.

File: game.go [hunk 5]
```diff
@@ -25,8 +25,42 @@ type GameListResult struct {
...
+        switch d {
+        case Forward:
+                next = pager.Limit + pager.From
```
Comment (ttaylorr) [inline game.go]: If you go with my comment above, this becomes a `:=`

Comment (JamyDev) [inline game.go]: backward case would still need a `var next int` so I think it's clearer to just do it at the top.

**Comment by JamyDev** [comment]
Did a `go fmt` @ttaylorr 
I didn't know go used tabs as indent :P

**Comment by MattJustMatt** [comment]
@JamyDev I approved Coveralls

**Comment by JamyDev** [comment]
Yes, however I can't seem to add the repo to it :/

**Comment by MattJustMatt** [comment]
Can I get that fixed for you, if so how? @JamyDev

**Comment by JamyDev** [comment]
I think I got dis

**Comment by JamyDev** [comment]
Green skies WOOP

**Comment by ttaylorr** [comment]
Looks good to me :+1:

**Comment by connor4312** [comment]
Looks good

**Pull request merged**